File: examples/python/durable_eviction/capacity_worker.py
"""
Dedicated worker for capacity-eviction e2e tests.

Runs with durable_slots=1 so that a single waiting durable task triggers
capacity pressure and gets evicted (even with ttl=None).
"""

from __future__ import annotations

from examples.durable_eviction.worker import capacity_evictable_sleep, hatchet


def main() -> None:
    worker = hatchet.worker(
        "capacity-eviction-worker",
        durable_slots=1,
        workflows=[capacity_evictable_sleep],
    )
    worker.start()


if __name__ == "__main__":
    main()

File: examples/typescript/durable_eviction/capacity-worker.ts
/**
 * Dedicated worker for capacity-eviction e2e tests.
 *
 * Runs with durableSlots=1 so that a single waiting durable task triggers
 * capacity pressure and gets evicted (even with ttl=undefined).
 */
import { hatchet } from '../hatchet-client';
import { capacityEvictableSleep } from './workflow';

async function main() {
  const worker = await hatchet.worker('capacity-eviction-worker', {
    durableSlots: 1,
    workflows: [capacityEvictableSleep],
  });

  await worker.start();
}

if (require.main === module) {
  main();
}
